src/data_access/FirebaseRoomsDataAccessObject.java

```diff
@@ -4,80 +4,221 @@
 import entities.auth.User;
 import entities.rooms.Message;
 import entities.rooms.Room;
+import java.io.IOException;
 import java.time.Instant;
-import java.util.ArrayList;
-import java.util.List;
-import use_case.rooms.Response;
+import java.util.*;
+import okhttp3.*;
+import org.json.JSONException;
+import org.json.JSONObject;
+import use_case.login.LoginUserDataAccessInterface;
 import use_case.rooms.RoomsDataAccessInterface;
 
-// TODO: You need to split this into multiple DAOs
 public class FirebaseRoomsDataAccessObject implements RoomsDataAccessInterface {
+
+  private final OkHttpClient client;
+
+  public FirebaseRoomsDataAccessObject(OkHttpClient client) {
+    this.client = client;
+  }
+
   @Override
-  public Response<List<Message>> loadMessages(Room room, User user) {
-    // TODO: You should get the messages corresponding to room
-    DisplayUser dummy_display_user = new DisplayUser("foo", "bar");
-    Instant timestamp = Instant.now();
-    Message message =
-        new Message(timestamp, "This is a test message to load.", dummy_display_user.getUid());
-
-    List<Message> messages = new ArrayList<>();
-    messages.add(message);
-
-    // Uncomment this to see what happens when you fail to get messages:
-    // Response<List<Messages>> response = new Response<List<Messages>>(null);
-    // response.setError("Failed to retrieve messages.");
-    // return response;
-
-    return new Response<>(messages);
+  public List<String> getAvailableRoomIds(User user) {
```

Comment:
No more `Response` pattern?

Reply:
I had already written the Room DAO before I merged your PR, so I had already been using the Exception pattern